File: StressTracker5001Server/DTOs/Card/CardDto.cs
namespace StressTracker5001Server.DTOs.Card
{
    public class CardDto
    {
        public required int Id { get; set; }
        public required int ColumnId { get; set; }
        public required string Title { get; set; }
        public required string Description { get; set; }
        public required int Position { get; set; }
        public DateTime? DueDate { get; set; }
        public required int CreatedById { get; set; }
        public required DateTime CreatedAt { get; set; }
        public required DateTime UpdatedAt { get; set; }
        public List<int> Tags { get; set; } = new();
        public List<CardAssignmentDto> Assignments { get; set; } = new();
    }
}

File: StressTracker5001App/src/dto/card.dto.ts
import type { AttachmentDto } from "./attachment.dto";
import type { CardAssignmentDto } from "./card-assignment.dto";
import type { UserDto } from "./user.dto";

export interface CardDetailsDto {
  id: number;
  columnId: number;
  title: string;
  description: string;
  position: number;
  dueDate: string | null;
  createdById: number;
  tags: number[];
  assignments: CardAssignmentDto[];
  createdBy: UserDto;
  attachments: AttachmentDto[];
  createdAt: string;
  updatedAt: string;
}

export interface CardDto {
  id: number;
  columnId: number;
  title: string;
  description: string;
  position: number;
  dueDate: string | null;
  createdById: number;
  createdAt: string;
  updatedAt: string;
  tags: number[];
  assignments: CardAssignmentDto[];
}

export interface CardCreateDto {
  title: string;
  description?: string;
  dueDate?: string | null;
}

export interface CardMoveDto {
  newPosition: number;
  newColumnId: number;
}

export interface CardAssignTagsDto {
  tags: number[];
}

export interface CardUpdateDto {
  title?: string;
  description?: string;
  dueDate?: string | null;
}
